hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/dstable/DistributionAddUpdateWindowLayout.java

```diff
@@ -176,31 +174,33 @@ private void addNewDistribution() {
             distributionSetTable.setValue(Sets.newHashSet(newDist.getId()));
         }
 
-        private boolean isDuplicate() {
-            final String name = distNameTextField.getValue();
-            final String version = distVersionTextField.getValue();
+        @Override
+        public boolean canWindowSaveOrUpdate() {
+            return !isDuplicate(null);
+        }
+    }
 
-            final Optional<DistributionSet> existingDs = distributionSetManagement.getByNameAndVersion(name, version);
-            /*
-             * Distribution should not exists with the same name & version.
-             * Display error message, when the "existingDs" is not null and it
-             * is add window (or) when the "existingDs" is not null and it is
-             * edit window and the distribution Id of the edit window is
-             * different then the "existingDs"
-             */
-            if (existingDs.isPresent() && !existingDs.get().getId().equals(editDistId)) {
-                distNameTextField.addStyleName(SPUIStyleDefinitions.SP_TEXTFIELD_LAYOUT_ERROR_HIGHTLIGHT);
-                distVersionTextField.addStyleName(SPUIStyleDefinitions.SP_TEXTFIELD_LAYOUT_ERROR_HIGHTLIGHT);
-                notificationMessage.displayValidationError(i18n.getMessage("message.duplicate.dist",
-                        new Object[] { existingDs.get().getName(), existingDs.get().getVersion() }));
-
-                return true;
-            }
+    private boolean isDuplicate(final Long editDistId) {
+        final String name = distNameTextField.getValue();
+        final String version = distVersionTextField.getValue();
 
-            return false;
+        final Optional<DistributionSet> existingDs = distributionSetManagement.getByNameAndVersion(name, version);
+        /*
+         * Distribution should not exists with the same name & version. Display
+         * error message, when the "existingDs" is not null and it is add window
+         * (or) when the "existingDs" is not null and it is edit window and the
+         * distribution Id of the edit window is different then the "existingDs"
+         */
+        if (existingDs.isPresent() && !existingDs.get().getId().equals(editDistId)) {
+            distNameTextField.addStyleName(SPUIStyleDefinitions.SP_TEXTFIELD_LAYOUT_ERROR_HIGHTLIGHT);
+            distVersionTextField.addStyleName(SPUIStyleDefinitions.SP_TEXTFIELD_LAYOUT_ERROR_HIGHTLIGHT);
+            notificationMessage.displayValidationError(i18n.getMessage("message.duplicate.dist",
+                    new Object[] { existingDs.get().getName(), existingDs.get().getVersion() }));
 
+            return true;
         }
 
+        return false;
```

Comment:
You could avoid the get() calls on the Optional like this:

        return existingDs.filter(ds -> ds.getId().equals(editDistId)).map(ds -> {
            distNameTextField.addStyleName(SPUIStyleDefinitions.SP_TEXTFIELD_LAYOUT_ERROR_HIGHTLIGHT);
          distVersionTextField.addStyleName(SPUIStyleDefinitions.SP_TEXTFIELD_LAYOUT_ERROR_HIGHTLIGHT);
            notificationMessage.displayValidationError(i18n.getMessage("message.duplicate.dist",
                    new Object[] { existingDs.get().getName(), existingDs.get().getVersion() }));
            return true;
        }).orElse(false);